coalib/parsing/filters/CanDetectFilter.py

```diff
@@ -0,0 +1,31 @@
+def can_detect_filter(bears, detects):
+    if detects:
+        bears = filter_bears_by_can_detects(bears, detects)
+    return bears
+
+
+def filter_bears_by_can_detects(bears, detects):
+    local_bears, global_bears = bears
+    local_bears = filter_section_bears_by_can_detects(local_bears, detects)
+    global_bears = filter_section_bears_by_can_detects(
```

Comment:
How about creating decorator for this ? And it will take care of this repetition.